module-03-react/src/components/Gallery.jsx

```diff
@@ -0,0 +1,146 @@
+import React, {
+    useEffect,
+    useState,
+    useImperativeHandle,
+    forwardRef,
+} from "react";
+import axios from "axios";
+import "/src/styles/components.css";
+import Card from "./Card.jsx";
+
+const baseURL = "https://swapi.dev/api/";
+
+const Gallery = forwardRef(
+    ({ setImg, search, setSearch, url, setOpenModal, setChoosedCard }, ref) => {
+        const [cards, setCards] = useState([]);
+        const [allResults, setAllResults] = useState([]);
+        const [page, setPage] = useState(1);
+
+        const [loading, setLoading] = useState(false);
+        const [error, setError] = useState(null);
+
+        const bottomButton = document.querySelector(".bottom__btn");
```

Comment:
Accessing the DOM directly with `document.querySelector(".bottom__btn")` bypasses React's Virtual DOM, creating potential synchronization issues and side effects. This is an anti-pattern in React as it operates outside React's rendering lifecycle and component state management. Instead, use React refs to access DOM elements when necessary